x/pairing/keeper/msg_server_relay_payment.go

```diff
@@ -267,7 +267,21 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
 
 		// if this returns an error it means this is legacy consumer
 		if !legacy {
-			err = k.projectsKeeper.AddComputeUnitsToProject(ctx, clientAddr.String(), uint64(relay.Epoch), relay.CuSum)
+			project, _, err := k.projectsKeeper.GetProjectForDeveloper(ctx, clientAddr.String(), uint64(relay.Epoch))
```

Comment:
all of this needs to be in getPairing function

Reply:
why?
Assuming you mean the whole code block under the !legacy condition, this code block adds CU to a project according to the relay payment message. It's unrelated to getPairing

Reply:
relaypayment callse verifyPairing that calls getPairing, this is the only place these checks needs to be done.
i am talking about the code regarding the project check